runtime/marshal.js

```diff
@@ -337,10 +337,46 @@ function caml_marshal_data_size (s, ofs) {
   return (get32(s, ofs + 4));
 }
 
+//Provides: MlObjectTable
+var MlObjectTable;
+if (typeof joo_global_object.WeakMap === 'undefined') {
+  MlObjectTable = function() {
```

Comment:
In the current test setup this code is never tested because the condition is always `false`. 😞

Reply:
I've tested the code manually.